flowmachine/flowmachine/features/subscriber/visited_most_days.py

```diff
@@ -0,0 +1,138 @@
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+
+from typing import List, Optional, Tuple
+from flowmachine.core import Query, make_spatial_unit
+from flowmachine.core.spatial_unit import AnySpatialUnit
+from ..utilities.subscriber_locations import BaseLocation, SubscriberLocations
+from flowmachine.utils import standardise_date
+
+
+class VisitedMostDays(BaseLocation, Query):
+    """
+    Another type of reference location which localises subscribers to the
+    location they visited on most days from a span of days, and breaks ties
+    according to the number of events they had at each location.
+
+    Parameters
+    ----------
+    start : str
+        iso format date range for the beginning of the time frame,
+        e.g. 2016-01-01 or 2016-01-01 14:03:01
+    stop : str
+        As above
+    spatial_unit : flowmachine.core.spatial_unit.*SpatialUnit, default admin3
+        Spatial unit to which subscriber locations will be mapped. See the
+        docstring of make_spatial_unit for more information.
+    hours : tuple of int, default 'all'
+        Subset the result within certain hours, e.g. (4,17)
+        This will subset the query only with these hours, but
+        across all specified days. Or set to 'all' to include
+        all hours.
+    table : str, default 'all'
+        schema qualified name of the table which the analysis is
+        based upon. If 'all' it will use all tables that contain
+        location data.
+    subscriber_identifier : {'msisdn', 'imei'}, default 'msisdn'
+        Either msisdn, or imei, the column that identifies the subscriber.
+    subscriber_subset : str, list, flowmachine.core.Query, flowmachine.core.Table, default None
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+
+    Notes
+    -----
+    * A date without a hours and mins will be interpreted as
+      midnight of that day, so to get data within a single day
+      pass (e.g.) '2016-01-01', '2016-01-02'.
+
+    * Use 24 hr format!
+    """
+
+    def __init__(
+        self,
+        start_date,
+        end_date,
+        spatial_unit: Optional[AnySpatialUnit] = None,
+        hours: Optional[Tuple[int, int]] = None,
+        table="all",
+        subscriber_identifier="msisdn",
+        *,
+        ignore_nulls=True,
+        subscriber_subset=None,
+    ):
+        self.start_date = standardise_date(start_date)
+        self.end_date = standardise_date(end_date)
+        if spatial_unit is None:
+            self.spatial_unit = make_spatial_unit("admin", level=1)
+        else:
+            self.spatial_unit = spatial_unit
+        self.hours = hours
+        self.table = table
+        self.subscriber_identifier = subscriber_identifier
+        self.subscriber_locs = SubscriberLocations(
```

Comment:
I think I tend to lean _slightly_ towards the latter, because when working with the library you can then create one `SubscriberLocations` obj and pass it into multiple queries.

Reply:
I think you're right. It wasn't so obvious to me at first but I now refactored it and the code is now a lot shorter and cleaner too and for the usage if anything it made it easier.
Given I stole that code from MostFrequentLocations, perhaps we should think about changing that too at some point?!